docs/source/_static/rest-api.yml

```diff
@@ -1380,6 +1769,95 @@ components:
           description: |
             Additional information a deployment can attach to a service.
             JupyterHub does not use this field.
+
+    SharedServer:
+      description: Subset of Server model present in Share responses
+      type: object
+      properties:
+        name:
+          type: string
+          description: the server name. '' for the default server.
+        url:
+          type: string
+          description: the server's URL
+        ready:
+          type: boolean
+          description: whether the server is ready
+        active:
+          type: boolean
+          description: whether the server is active
+        user:
+          type: object
+          description: the server's owner
+          properties:
+            name:
+              type: string
+
+    Share:
+      description: |
+        A single sharing permission.
+        There is at most one of these objects per (server, user) or (server, group) combination.
+      type: object
+      properties:
+        server:
+          description: the server granting shared access
+          $ref: "#/components/schemas/SharedServer"
+        scopes:
+          description: the scopes granted by this Share
+          type: array
+          items:
+            type: string
+        group:
+          description:
+            the group being shared with (exactly one of 'user' or 'group'
+            will be non-null, the other will be null)
+          type: object
+          nullable: true
+          properties:
+            name:
+              type: string
+        user:
+          description:
+            the user being shared with (exactly one of 'user' or 'group'
+            will be non-null, the other will be null)
+          type: object
+          nullable: true
+          properties:
+            name:
+              type: string
```

Comment:
oneOf requires additional schemas for the mutually exclusive groups. I think this is clearer than adding the abstractions required to make the schema enforce mutual exclusivity, since it doesn't communicate it any better than the prose, and this is exclusively intended for docs generation.